cmd/kyma/provision/minikube/cmd.go

```diff
@@ -236,6 +237,15 @@ func (c *command) startMinikube() error {
 	if c.opts.VMDriver == vmDriverHyperv {
 		startCmd = append(startCmd, "--hyperv-virtual-switch="+c.opts.HypervVirtualSwitch)
 	}
+
+	if c.opts.UseVPNKitSock {
```

Comment:
The "whoami" command is OS specific, the pathToVPNKitSock is OS specific, the --hyperkit-vpnkit-sock flag is MacOS specific as hyperkit is not supported by other platforms. I think you need to make it all MacOS specific so that it is not available at all for other platforms.

Reply:
fixed.

Reply:
This is also my thought, the code will be compiled into all OS versions as it is now. My idea when we discussed this was to move the OS specific logic out so that it only compiles in Mac OSX.

How it is currently will not crash because of the condition being only true on Mac OSX but it is not the best way to achieve it. As much as possible should be on the OS specific code, not just a boolean. 

Also, the boolean `allowVPNSock` can be completely replaced by `runtime.GOOS == "darwin"`) so it is not necessary at all.